csharp/extractor/Semmle.Extraction.CSharp/Entities/Modifier.cs

```diff
@@ -97,7 +100,9 @@ private static void ExtractNamedTypeModifiers(Context cx, TextWriter trapFile, I
 
         public static void ExtractModifiers(Context cx, TextWriter trapFile, IEntity key, ISymbol symbol)
         {
-            HasAccessibility(cx, trapFile, key, symbol.DeclaredAccessibility);
+            if (symbol.Kind != SymbolKind.NamedType || !((INamedTypeSymbol)symbol).IsFileLocal)
```

Comment:
It's not clear to me why this new check was introduced here; perhaps a comment explaining the logic in this method would be helpful?

Reply:
That is a good point! I will add a comment.
When `IsFileLocal` is true it means that the type has declared scope `file`.
It turns out that the symbol pertaining to `C1` in the example below has declared accessibility *internal*, which I don't think we should extract. There are similar incorrect extractions for explicit interface implementations, which is then "fixed" in the QL library layer (which leads to unwanted complexity), but I prefer that we avoid extracting the "internal" modifier in the first place.

```csharp
file class C1 { }
```

Reply:
Makes sense, and the `symbol.Kind != SymbolKind.NamedType` condition is there to check that this situation applies / for the cast to succeed. Thanks!